src/components/Table/TableFilterButtons/buildFilterItems.tsx

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import type {ReactNode} from 'react';
+import type {PopoverComponentProps} from '@components/Search/FilterDropdowns/DropdownButton';
+import MultiSelectPopup from '@components/Search/FilterDropdowns/MultiSelectPopup';
+import SingleSelectPopup from '@components/Search/FilterDropdowns/SingleSelectPopup';
+import type {FilterConfig, FilterConfigEntry} from '@components/Table/types';
+
+type FilterButtonItem = {
+    key: string;
+    label: string;
+    value: string | string[] | null;
+    PopoverComponent: (props: PopoverComponentProps) => React.ReactNode;
+};
+
+function buildFilterItems(
+    filterConfigs: FilterConfig | undefined,
+    filters: Record<string, unknown>,
+    setFilter: (key: string, value: unknown) => void,
+    filtersLabel: string,
+): FilterButtonItem[] {
+    if (!filterConfigs) {
+        return [];
+    }
+
+    return Object.keys(filterConfigs).map((filterKey) => {
+        const filterConfig = filterConfigs[filterKey];
+        const currentFilterValue = filters[filterKey];
+        const displayValue = getDisplayValue(filterConfig, currentFilterValue);
+
+        const PopoverComponent = createPopoverComponent(filterKey, filterConfig, currentFilterValue, setFilter);
+
+        // Determine the label to display
+        let label: string;
+        if (filterConfig.filterType === 'multi-select') {
+            // For multi-select: show selected values joined, or translated "Filters" if nothing selected
+            if (Array.isArray(displayValue) && displayValue.length > 0) {
+                label = displayValue.join(', ');
+            } else {
+                label = filtersLabel;
+            }
+
+            return {
+                key: filterKey,
+                label,
+                value: null,
+                PopoverComponent,
+            };
+        }
+
+        // For single-select: show display value, or default option label, or filterKey
+        if (displayValue && !Array.isArray(displayValue)) {
+            label = displayValue;
+        } else if (filterConfig.default) {
+            // Find the default option label
+            const defaultOption = filterConfig.options.find((opt) => opt.value === filterConfig.default);
+            label = defaultOption?.label ?? filterKey;
+        } else {
+            label = filterKey;
+        }
+
+        return {
+            key: filterKey,
+            label,
+            value: null,
+            PopoverComponent,
+        };
+    });
+}
+
+function createPopoverComponent(
+    filterKey: string,
+    filterConfig: FilterConfigEntry,
+    currentFilterValue: unknown,
+    setFilter: (key: string, value: unknown) => void,
+): (props: PopoverComponentProps) => ReactNode {
+    if (filterConfig.filterType === 'multi-select') {
```

Comment:
Rather than having one `FilterButtonItem` that has different filter types, it might be better to have two different components:

1. FilterButtonSingleSelectItem
2. FilterButtonMultiSelectItem

and then DRY up the logic between them